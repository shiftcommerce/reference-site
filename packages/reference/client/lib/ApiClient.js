// Libraries
import qs from 'qs'
import axios from 'axios'

class ApiClient {
  constructor (options = {}) {
    this.client = axios.create({
      baseURL: process.env.API_HOST_PROXY
    })
  }

  async read (endpoint, queryObject = {}, options = {}) {
    const formattedEndpoint = this.encodeParams(endpoint, queryObject)

    try {
      const response = await this.client.get(formattedEndpoint)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('API CLIENT: Error while fetching data', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  async post (endpoint, body = {}, options = {}) {
    try {
      const response = await this.client.post(endpoint, body)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('Error while posting data', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  encodeParams (endpoint, queryObject = {}) {
    if (Object.keys(queryObject).length > 0) {
      const query = qs.stringify(queryObject)
      endpoint = `${endpoint}?${query}`
    }
    return endpoint
  }
}

export default ApiClient
