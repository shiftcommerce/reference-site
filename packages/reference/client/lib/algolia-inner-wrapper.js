// Lib
import InitialPropsDelegator from './initial-props-delegator'
import getConfig from 'next/config'
import { Configure } from 'react-instantsearch/dom'

// Components
import { InstantSearch } from '../components/search/algolia/instant-search'

const {
  publicRuntimeConfig: {
    ALGOLIA_API_KEY,
    ALGOLIA_APP_ID,
    ALGOLIA_INDEX_NAME
  }
} = getConfig()

export default function algoliaInnerWrapper (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const { resultsState, onSearchStateChange, searchState, ...otherProps } = this.props
      return (
        <InstantSearch
          appId={ALGOLIA_APP_ID}
          apiKey={ALGOLIA_API_KEY}
          indexName={ALGOLIA_INDEX_NAME}
          resultsState={resultsState}
          onSearchStateChange={onSearchStateChange}
          searchState={searchState}
        >
          <Configure {...searchState.configure} />
          <Component {...otherProps} />
        </InstantSearch>
      )
    }
  }
}
