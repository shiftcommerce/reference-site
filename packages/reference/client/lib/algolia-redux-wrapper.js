// Libraries
import withReduxStore from './with-redux-store'
import { Provider } from 'react-redux'
import { Configure } from 'react-instantsearch/dom'
import Router from 'next/router'
import qs from 'qs'
import getConfig from 'next/config'

// Lib
import InitialPropsDelegator from './initial-props-delegator'
import buildSearchStateForURL from '../lib/build-search-state-for-url'

// Components
import ConnectedLayout from '../components/layout'
import { InstantSearch, findResultsState } from '../components/search/algolia/instant-search'

const {
  publicRuntimeConfig: {
    ALGOLIA_API_KEY,
    ALGOLIA_APP_ID,
    ALGOLIA_INDEX_NAME,
    ALGOLIA_RESULTS_PER_PAGE
  }
} = getConfig()

export default function (ConnectedComponent, Page) {
  return algoliaOuterWrapper(
    withReduxStore(
      withProvider(
        algoliaInnerWrapper(
          withLayout(
            ConnectedComponent
          )
        )
      )
    ),
    Page
  )
}

function algoliaOuterWrapper (NextWrapper, Page) {
  return class extends InitialPropsDelegator(NextWrapper) {
    constructor (props) {
      super(props)
      this.onSearchStateChange = this.onSearchStateChange.bind(this)
    }

    static async getInitialProps (appContext) {
      const { query } = appContext

      // Get initial props from the parent
      const appProps = await Object.getPrototypeOf(this).getInitialProps(appContext)

      let searchState = query
      if (Page.buildAlgoliaStates) {
        searchState = Page.buildAlgoliaStates(appContext)
      }

      let resultsState = {
        _originalResponse: {
          results: [{}]
        }
      }
      if (Page.algoliaEnabled && Page.algoliaEnabled()) {
        resultsState = await findResultsState(NextWrapper, { searchState, ...appProps })
      }

      return {
        ...appProps,
        resultsState,
        searchState
      }
    }

    updateAfter () {
      // Delegate to custom implementation if the page provides one
      if (Page.updateAfter) {
        return Page.updateAfter()
      }

      // Default implementation
      return 700
    }

    onSearchStateChange (searchState) {
      // Delegate to custom implementation if the page provides one
      if (Page.onSearchStateChange) {
        return Page.onSearchStateChange.call(this, searchState)
      }

      // Default implementation
      clearTimeout(this.debouncedSetState)
      this.debouncedSetState = setTimeout(() => {
        const href = this.searchStateToUrl(searchState)
        Router.push(href, href, { shallow: true })
      }, this.updateAfter())
      this.setState({ searchState })
    }

    searchStateToUrl (searchState) {
      // Delegate to custom implementation if the page provides one
      if (Page.searchStateToUrl) {
        return Page.searchStateToUrl.call(this, searchState)
      }

      // Default implementation
      const urlSearchState = buildSearchStateForURL(searchState)
      // Build the query string and append it to search path
      return searchState.query ? `/search?${qs.stringify(urlSearchState)}` : '/search'
    }

    componentDidMount () {
      return Page.algoliaComponentDidMount && Page.algoliaComponentDidMount.call(this)
    }

    componentWillReceiveProps (nextProps) {
      return Page.algoliaComponentWillReceiveProps && Page.algoliaComponentWillReceiveProps.call(this, nextProps)
    }

    render () {
      const { resultsState, searchState, ...appProps } = this.props
      return (
        <NextWrapper
          resultsState={resultsState}
          onSearchStateChange={this.onSearchStateChange}
          searchState={
            this.state && this.state.searchState
              ? this.state.searchState
              : searchState
          }
          {...appProps}
        />
      )
    }
  }
}

function withProvider (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const { reduxStore, ...otherProps } = this.props
      return (
        <Provider store={reduxStore}>
          <Component {...otherProps} />
        </Provider>
      )
    }
  }
}

function algoliaInnerWrapper (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const { resultsState, onSearchStateChange, searchState, ...otherProps } = this.props
      // We should always configure this in the state to prevent a onSearchStateChange first render
      searchState.configure = { ...searchState.configure, hitsPerPage: ALGOLIA_RESULTS_PER_PAGE }
      return (
        <InstantSearch
          appId={ALGOLIA_APP_ID}
          apiKey={ALGOLIA_API_KEY}
          indexName={ALGOLIA_INDEX_NAME}
          resultsState={resultsState}
          onSearchStateChange={onSearchStateChange}
          searchState={searchState}
        >
          <Configure {...searchState.configure} />
          <Component searchState={ searchState } {...otherProps} />
        </InstantSearch>
      )
    }
  }
}

function withLayout (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      return (
        <ConnectedLayout>
          <Component {...this.props} />
        </ConnectedLayout>
      )
    }
  }
}
