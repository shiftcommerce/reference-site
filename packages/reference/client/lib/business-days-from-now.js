// Libraries
import moment from 'moment-business-days'

moment.locale('en-gb')

export function businessDaysFromNow (days) {
  return moment(new Date()).businessAdd(days)
}
