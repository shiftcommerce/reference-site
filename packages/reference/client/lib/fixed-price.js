export function fixedPrice (price) {
  return (price).toFixed(2)
}
