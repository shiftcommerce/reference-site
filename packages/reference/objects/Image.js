import { Component } from 'react'
import classNames from 'classnames'

class Image extends Component {
  styles () {
    let basicStyles = {
      height: this.props.height,
      width: this.props.width
    }

    if (basicStyles.height < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.height * 19) / 20) }
    } else if (basicStyles.width < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.width * 19) / 20) }
    } else {
      return { ...basicStyles, backgroundSize: 150 }
    }
  }

  render () {
    let {
      src,
      className
    } = this.props

    if (src) {
      return <img {...this.props} />
    } else {
      return <div className={classNames('c-dummy_image', className)} style={this.styles()} />
    }
  }
}

export default Image
