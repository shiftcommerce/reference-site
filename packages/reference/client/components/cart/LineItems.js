// Libraries
import { Component } from 'react'
import Link from 'next/link'
import { fixedPrice } from '../../lib/fixedPrice'

// Objects
import Image from '../../objects/Image'

class LineItems extends Component {
  // TODO: extract this out into it's own service class
  renderOptions (lineItem) {
    let availableQuantity = Array.from(Array(Math.min(lineItem.stockAvailableLevel, 11)).keys())
    availableQuantity.shift()

    return <select value={lineItem.quantity} onChange={this.props.updateQuantity} data-variant={lineItem.sku}>
      {
        availableQuantity.map(index => {
          return <option key={index} value={index} aria-setsize={index} aria-posinset={index + 1}>
            {index}
          </option>
        })
      }
    </select>
  }

  renderButtonsAndTotal (lineItem) {
    return <>
      <div className='o-grid-item5__lineitems'>
        <div className='c-line-items__buttons'>
          <div className='c-line-items__anchor'>
            <a className='c-line-items__delete' data-variant={lineItem.sku} onClick={this.props.deleteItem} >
              Delete
            </a>
          </div>
        </div>
        <div className='c-line-items__total-wrapper'>
          <a className='c-line-items__total'>&pound;{fixedPrice(lineItem.price * lineItem.quantity)}</a>
        </div>
      </div>
    </>
  }

  renderTitle (lineItem) {
    return <>
      <div className='o-grid-item2__lineitems'>
        <div className='c-line-items__details'>
          <h4 className='c-line-items__details-title  u-bold'>
            {lineItem.title}
          </h4>
          <div className='c-line-items__details-sku'>
            <span>
              {lineItem.sku}
            </span>
          </div>
        </div>
      </div>
    </>
  }

  renderParams (lineItem) {
    return <>
      <div className='o-grid-item3__lineitems'>
        <div className='c-line-items__colour  u-bold'>
          <span>Colour</span>
        </div>
        <div className='c-line-items__size  u-bold'>
          <span>Option</span>
        </div>
        <div className='c-line-items__quantity u-bold'>
          <span>Quantity</span>
        </div>
      </div>
      <div className='o-grid-item4__lineitems'>
        <span className='c-line-items__space'>Grey</span>
        <span className='c-line-items__space'>{lineItem.variant}</span>
        <span className='c-line-items__space'>{ this.renderOptions(lineItem) }</span>
      </div>
    </>
  }

  renderLineItems () {
    const cartData = this.props.cart.lineItems.map((lineItem, index) =>
      <div className='o-grid-container__lineitems' key={lineItem.slug}>
        <div className='o-grid-item1__lineitems'>
          <Link href={`/slug?slug=${lineItem.slug}`} as={`${lineItem.canonical_path}`} key={lineItem.slug}>
            <Image className='c-line-items__image' src={lineItem.imageUrl} alt={lineItem.title} key={lineItem.slug} aria-label={lineItem.title} />
          </Link>
        </div>
        { this.renderTitle(lineItem) }
        { this.renderParams(lineItem) }
        { this.renderButtonsAndTotal(lineItem) }
      </div>
    )
    return cartData
  }

  render () {
    const cart = this.props.cart

    if (cart.totalQuantity === 0) {
      return null
    } else {
      return <div className='c-line-items'>
        { this.renderLineItems() }
      </div>
    }
  }
}

export default LineItems
