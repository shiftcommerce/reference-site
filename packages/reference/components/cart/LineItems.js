import { Component } from 'react'
import Link from 'next/link'

// Objects
import Image from '../../objects/Image'

export default class LineItems extends Component {
  renderOptions (lineItem) {
    return <select value={lineItem.quantity} onChange={this.props.updateQuantity} data-variant={lineItem.sku}>
      {
        [ ...Array(Math.max(lineItem.quantity, 11)) ].map((key, index) =>
          <option value={index} key={index} aria-setsize={index} aria-posinset={index + 1}>
            { index }
          </option>
        )
      }
    </select>
  }

  renderLineItems () {
    const cartData = this.props.cart.lineItems.map((lineItem, index) =>
      <tr key={index} className='c-line-items__column'>
        <td aria-label={lineItem.title}>
          <Link href={`/products/${lineItem.productSku}`}>
            <a>
              <Image src={lineItem.imageUrl} alt={lineItem.title} className='c-line-items__image' aria-label={lineItem.title} />
              { lineItem.title }
            </a>
          </Link>
        </td>
        <td>
          { lineItem.size }
        </td>
        <td>
          { this.renderOptions(lineItem) }
        </td>
        <td>
          -&pound;{ (lineItem.discount || 0) * lineItem.quantity }
        </td>
        <td>
          &pound;{ lineItem.price * lineItem.quantity }
        </td>
      </tr>
    )
    return cartData
  }

  render () {
    const cart = this.props.cart

    if (cart.lineItems.length === 0) {
      return null
    } else {
      return <table className='c-line-items'>
        <thead>
          <tr className='c-line-items__header'>
            <th scope='col'>
              Your Items
            </th>
            <th scope='col'>
              Size
            </th>
            <th scope='col'>
              Quantity
            </th>
            <th scope='col'>
              You save
            </th>
            <th scope='col'>
              Subtotal
            </th>
          </tr>
        </thead>
        <tbody>
          { this.renderLineItems() }
        </tbody>
      </table>
    }
  }
}
