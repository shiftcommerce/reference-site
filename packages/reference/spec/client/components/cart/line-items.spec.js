import LineItems from '../../../../client/components/cart/line-items'
import Image from '../../../../client/objects/image'

import { fixedPrice } from '../../../../client/lib/fixed-price'

test('renders line items correctly, on initial load', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        title: 'test',
        price: fixedPrice(10.0),
        discount: 0,
        quantity: 2,
        sku: '123',
        imageUrl: 'https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12917/S2623404_C146_Main.jpg',
        size: 'size - 8',
        stockAvailableLevel: '1000',
        variant: '38 Waist 31 Leg',
        canonicalPath: '1',
        slug: '1'
      }
    ]
  }

  const updateQuantity = () => {}

  // act
  const wrapper = shallow(
    <LineItems cart={cart} updateQuantity={updateQuantity} />
  )

  // assert
  const lineItem = cart.lineItems[0]
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(lineItem.title)
  expect(wrapper).toContainReact(<Image src={lineItem.imageUrl} alt={lineItem.title} className='c-line-items__image' aria-label={lineItem.title} />)
  expect(wrapper.find('select').props().value).toBe(lineItem.quantity)
  expect(wrapper).toIncludeText(lineItem.sku)
  expect(wrapper).toIncludeText(lineItem.variant)
  expect(wrapper).toIncludeText((lineItem.price * lineItem.quantity) - (lineItem.discount))
})

test('trigger updateQuantity function, on change of line item quantity', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        title: 'test',
        price: fixedPrice(10.0),
        discount: 0,
        quantity: 2,
        sku: '123',
        imageUrl: 'https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12917/S2623404_C146_Main.jpg',
        size: 'size - 8',
        stockAvailableLevel: '1000',
        variant: '38 Waist 31 Leg',
        canonicalPath: '1',
        slug: '1'
      }
    ]
  }

  const updateQuantity = jest.fn()

  // act
  const wrapper = shallow(
    <LineItems cart={cart} updateQuantity={updateQuantity} />
  )

  // assert
  wrapper.find('select').simulate('change', { target: { value: 3 } })
  expect(wrapper).toMatchSnapshot()
  expect(updateQuantity).toHaveBeenCalled()
})
