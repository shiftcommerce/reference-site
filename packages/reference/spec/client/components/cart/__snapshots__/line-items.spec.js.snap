// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders line items correctly, on initial load 1`] = `
<div
  className="c-line-items"
>
  <div
    className="c-line-items__sections"
    key="1"
  >
    <div
      className="c-line-items__images"
    >
      <Link
        href="/slug?slug=1"
        key="1"
      >
        <Image
          className="c-line-items__image"
          key="1"
          src="https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12917/S2623404_C146_Main.jpg"
        />
      </Link>
    </div>
    <div
      className="c-line-items__information"
    >
      <div
        className="c-line-items__title"
      >
        <div
          className="c-line-items__details"
        >
          <h4
            className="c-line-items__details-title  u-bold"
          >
            Test
          </h4>
          <div
            className="c-line-items__details-sku"
          >
            <span>
              123
            </span>
          </div>
        </div>
      </div>
      <React.Fragment>
        <div
          className="c-line-items__colour"
        >
          <div
            className="c-line-items__param  u-bold"
          >
            <span>
              Colour
            </span>
          </div>
        </div>
        <div
          className="c-line-items__quantity"
        >
          <div
            className="c-line-items__param u-bold"
          >
            <span>
              Quantity
            </span>
          </div>
        </div>
        <div
          className="c-line-items__param  c-line-items__colour-selected"
        >
          <span>
            Grey
          </span>
        </div>
        <div
          className="c-line-items__param  c-line-items__quantity-selected"
        >
          <span>
            <select
              data-variant="123"
              onChange={[Function]}
              value={2}
            >
              <option
                aria-posinset={2}
                aria-setsize={1}
                key="1"
                value={1}
              >
                1
              </option>
              <option
                aria-posinset={3}
                aria-setsize={2}
                key="2"
                value={2}
              >
                2
              </option>
              <option
                aria-posinset={4}
                aria-setsize={3}
                key="3"
                value={3}
              >
                3
              </option>
              <option
                aria-posinset={5}
                aria-setsize={4}
                key="4"
                value={4}
              >
                4
              </option>
              <option
                aria-posinset={6}
                aria-setsize={5}
                key="5"
                value={5}
              >
                5
              </option>
              <option
                aria-posinset={7}
                aria-setsize={6}
                key="6"
                value={6}
              >
                6
              </option>
              <option
                aria-posinset={8}
                aria-setsize={7}
                key="7"
                value={7}
              >
                7
              </option>
              <option
                aria-posinset={9}
                aria-setsize={8}
                key="8"
                value={8}
              >
                8
              </option>
              <option
                aria-posinset={10}
                aria-setsize={9}
                key="9"
                value={9}
              >
                9
              </option>
              <option
                aria-posinset={11}
                aria-setsize={10}
                key="10"
                value={10}
              >
                10
              </option>
            </select>
          </span>
        </div>
      </React.Fragment>
      <React.Fragment>
        <div
          className="c-line-items__remove"
        >
          <div
            className="c-line-items__delete"
          >
            <a
              className="c-line-items__delete-button"
              data-variant="123"
            >
              Delete
            </a>
          </div>
        </div>
        <div
          className="c-line-items__amount"
        >
          <a
            className="c-line-items__total"
          >
            £
            20.00
          </a>
        </div>
      </React.Fragment>
    </div>
  </div>
</div>
`;

exports[`trigger updateQuantity function, on change of line item quantity 1`] = `
<div
  className="c-line-items"
>
  <div
    className="c-line-items__sections"
    key="1"
  >
    <div
      className="c-line-items__images"
    >
      <Link
        href="/slug?slug=1"
        key="1"
      >
        <Image
          className="c-line-items__image"
          key="1"
          src="https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12917/S2623404_C146_Main.jpg"
        />
      </Link>
    </div>
    <div
      className="c-line-items__information"
    >
      <div
        className="c-line-items__title"
      >
        <div
          className="c-line-items__details"
        >
          <h4
            className="c-line-items__details-title  u-bold"
          >
            Test
          </h4>
          <div
            className="c-line-items__details-sku"
          >
            <span>
              123
            </span>
          </div>
        </div>
      </div>
      <React.Fragment>
        <div
          className="c-line-items__colour"
        >
          <div
            className="c-line-items__param  u-bold"
          >
            <span>
              Colour
            </span>
          </div>
        </div>
        <div
          className="c-line-items__quantity"
        >
          <div
            className="c-line-items__param u-bold"
          >
            <span>
              Quantity
            </span>
          </div>
        </div>
        <div
          className="c-line-items__param  c-line-items__colour-selected"
        >
          <span>
            Grey
          </span>
        </div>
        <div
          className="c-line-items__param  c-line-items__quantity-selected"
        >
          <span>
            <select
              data-variant="123"
              onChange={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "target": Object {
                          "value": 3,
                        },
                      },
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                }
              }
              value={2}
            >
              <option
                aria-posinset={2}
                aria-setsize={1}
                key="1"
                value={1}
              >
                1
              </option>
              <option
                aria-posinset={3}
                aria-setsize={2}
                key="2"
                value={2}
              >
                2
              </option>
              <option
                aria-posinset={4}
                aria-setsize={3}
                key="3"
                value={3}
              >
                3
              </option>
              <option
                aria-posinset={5}
                aria-setsize={4}
                key="4"
                value={4}
              >
                4
              </option>
              <option
                aria-posinset={6}
                aria-setsize={5}
                key="5"
                value={5}
              >
                5
              </option>
              <option
                aria-posinset={7}
                aria-setsize={6}
                key="6"
                value={6}
              >
                6
              </option>
              <option
                aria-posinset={8}
                aria-setsize={7}
                key="7"
                value={7}
              >
                7
              </option>
              <option
                aria-posinset={9}
                aria-setsize={8}
                key="8"
                value={8}
              >
                8
              </option>
              <option
                aria-posinset={10}
                aria-setsize={9}
                key="9"
                value={9}
              >
                9
              </option>
              <option
                aria-posinset={11}
                aria-setsize={10}
                key="10"
                value={10}
              >
                10
              </option>
            </select>
          </span>
        </div>
      </React.Fragment>
      <React.Fragment>
        <div
          className="c-line-items__remove"
        >
          <div
            className="c-line-items__delete"
          >
            <a
              className="c-line-items__delete-button"
              data-variant="123"
            >
              Delete
            </a>
          </div>
        </div>
        <div
          className="c-line-items__amount"
        >
          <a
            className="c-line-items__total"
          >
            £
            20.00
          </a>
        </div>
      </React.Fragment>
    </div>
  </div>
</div>
`;
